src/macros.jl

```diff
@@ -159,25 +159,56 @@ function parse_one_operator_constraint(_error::Function, args...)
     _unknown_constraint_expr(_error)
 end
 
+# When `_rewrite_expr` is called with an expression, it returns three things:
+# - code for parsing (which must be called first), if needed; otherwise, :().
+# - code for building the required constraints, if needed; otherwise, :().
+# - the symbol of the variable that replaces the expression (<: VariableRef).
+_rewrite_expr(_error::Function, ::Val{:call}, ::Val{OP}, args...) where OP = :(), :(), Expr(:call, OP, args...)
+_rewrite_expr(_error::Function, ::Val{HEAD}, args...) where HEAD = :(), :(), Expr(HEAD, args...)
+_rewrite_expr(_error::Function, e::Symbol) = :(), :(), e
+_rewrite_expr(_error::Function, e::Number) = :(), :(), e
+
+_rewrite_expr(_error::Function, e::Expr) = _rewrite_expr(_error, Val(e.head), e.args...)
+_rewrite_expr(_error::Function, head::Val{:call}, args...) =
+    _rewrite_expr(_error, Val(:call), Val(args[1]), args[2:end]...)
+
+function _rewrite_expr(_error::Function, ::Val{:call}, op::Union{Val{:+}, Val{:-}, Val{:*}, Val{:/}}, args...)
```

Comment:
Why have a method with `OP` above which does not rewrite recursively and this one which rewrite recursively ?

Reply:
The previous one is a fallback for unimplemented operators. I don't know if it's the most sensible default: I try to enable this code as rarely as possible, to avoid breaking things.